x-pack/plugins/security_solution/server/lib/telemetry/receiver.ts

```diff
@@ -278,15 +291,38 @@ export class TelemetryReceiver {
       body: {
         query: {
           bool: {
-            filter: [
-              { term: { 'alert.alertTypeId': 'siem.signals' } },
-              { term: { 'alert.params.immutable': true } },
+            must: [
+              {
+                bool: {
+                  filter: {
+                    terms: {
+                      'alert.alertTypeId': [
+                        SIGNALS_ID,
+                        EQL_RULE_TYPE_ID,
+                        ML_RULE_TYPE_ID,
+                        QUERY_RULE_TYPE_ID,
+                        SAVED_QUERY_RULE_TYPE_ID,
+                        INDICATOR_RULE_TYPE_ID,
+                        THRESHOLD_RULE_TYPE_ID,
+                      ],
+                    },
+                  },
+                },
+              },
+              {
+                bool: {
+                  filter: {
+                    terms: {
+                      'alert.params.immutable': ['true'],
```

Comment:
NOTE: One small change to the backport is that this is a string because of this error:
```json
info deleting 199 ts output directories
--
  | info Building TypeScript projects refs for tsconfig.refs.json...
  | info [tsc] > node_modules/typescript/bin/tsc -b tsconfig.refs.json --pretty
  | proc [tsc] x-pack/plugins/security_solution/server/lib/telemetry/receiver.ts:314:19 - error TS2322: Type '{ terms: { 'alert.params.immutable': true[]; }; }' is not assignable to type 'QueryDslQueryContainer \| QueryDslQueryContainer[] \| undefined'.
  | proc [tsc]   Types of property 'terms' are incompatible.
  | proc [tsc]     Type '{ 'alert.params.immutable': true[]; }' is not assignable to type 'QueryDslTermsQueryKeys \| { [property: string]: string[] \| number[] \| QueryDslTermsLookup; } \| undefined'.
  | proc [tsc]       Types of property ''alert.params.immutable'' are incompatible.
  | proc [tsc]         Type 'true[]' is not assignable to type 'string[] \| number[] \| QueryDslTermsLookup \| undefined'.
  | proc [tsc]           Type 'true[]' is not assignable to type 'string[]'.
  | proc [tsc]             Type 'boolean' is not assignable to type 'string'.
  | proc [tsc]
  | proc [tsc] 314                   filter: {
  | proc [tsc]                       ~~~~~~
  | proc [tsc]
  | proc [tsc]
  | proc [tsc] Found 1 error.
  | proc [tsc]
  | succ ts refs build successfully